src/Microsoft.Identity.Web/TokenAcquisition.cs

```diff
@@ -347,6 +351,19 @@ private async Task<IConfidentialClientApplication> BuildConfidentialClientApplic
             string authority;
             IConfidentialClientApplication app;
 
+            MicrosoftIdentityOptionsValidation microsoftIdentityOptionsValidation = new MicrosoftIdentityOptionsValidation();
+            if (microsoftIdentityOptionsValidation.ValidateClientSecret(_applicationOptions).Failed)
```

Comment:
Perhaps we can create a CCAOptionsValidation class that validates this. It would also be more extensible and we could get the failure message from the result itself.